zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/JobHandler.java

```diff
@@ -0,0 +1,148 @@
+  public List<String> generateIds(final Record<JobRecordValue> record) {
+    return List.of(String.valueOf(record.getKey()));
+  }
+
+  @Override
+  public JobEntity createNewEntity(final String id) {
+    return new JobEntity().setId(id);
+  }
+
+  @Override
+  public void updateEntity(final Record<JobRecordValue> record, final JobEntity entity) {
+
+    final JobRecordValue recordValue = record.getValue();
+    entity.setKey(record.getKey());
+    entity
+        .setPartitionId(record.getPartitionId())
+        .setProcessInstanceKey(recordValue.getProcessInstanceKey())
+        .setFlowNodeInstanceId(recordValue.getElementInstanceKey())
+        .setBpmnProcessId(recordValue.getBpmnProcessId())
+        .setProcessDefinitionKey(recordValue.getProcessDefinitionKey())
+        .setTenantId(recordValue.getTenantId())
+        .setType(recordValue.getType())
+        .setWorker(recordValue.getWorker())
+        .setState(record.getIntent().name())
+        .setRetries(recordValue.getRetries())
+        .setErrorMessage(recordValue.getErrorMessage())
+        .setErrorCode(recordValue.getErrorCode())
+        .setEndTime(DateUtil.toOffsetDateTime(Instant.ofEpochMilli(record.getTimestamp())))
+        .setCustomHeaders(recordValue.getCustomHeaders())
+        .setJobKind(recordValue.getJobKind().name())
+        .setFlowNodeId(recordValue.getElementId());
+
+    if (recordValue.getJobListenerEventType() != null) {
+      entity.setListenerEventType(recordValue.getJobListenerEventType().name());
+    }
+    final long jobDeadline = recordValue.getDeadline();
+    if (jobDeadline >= 0) {
+      entity.setDeadline(DateUtil.toOffsetDateTime(Instant.ofEpochMilli(jobDeadline)));
+    }
+
+    if (FAILED_JOB_EVENTS.contains(record.getIntent())) {
+      // set flowNodeId to null to not overwrite it (because zeebe puts an error message there)
+      entity.setFlowNodeId(null);
+      if (recordValue.getRetries() > 0) {
+        entity.setJobFailedWithRetriesLeft(true);
+      } else {
+        entity.setJobFailedWithRetriesLeft(false);
+      }
+    }
+  }
+
+  @Override
+  public void flush(final JobEntity jobEntity, final BatchRequest batchRequest) {
+    final Map<String, Object> updateFields = new HashMap<>();
+    if (jobEntity.getFlowNodeId() != null) {
+      updateFields.put(FLOW_NODE_ID, jobEntity.getFlowNodeId());
+    }
+    updateFields.put(JOB_WORKER, jobEntity.getWorker());
+    updateFields.put(JOB_STATE, jobEntity.getState());
+    updateFields.put(RETRIES, jobEntity.getRetries());
+    updateFields.put(ERROR_MESSAGE, jobEntity.getErrorMessage());
+    updateFields.put(ERROR_CODE, jobEntity.getErrorCode());
+    updateFields.put(TIME, jobEntity.getEndTime());
+    updateFields.put(CUSTOM_HEADERS, jobEntity.getCustomHeaders());
+    updateFields.put(JOB_DEADLINE, jobEntity.getDeadline());
+    updateFields.put(PROCESS_DEFINITION_KEY, jobEntity.getProcessDefinitionKey());
+    updateFields.put(BPMN_PROCESS_ID, jobEntity.getBpmnProcessId());
```

Comment:
❓ should we add `JobFailedWithRetriesLeft` to the updateFields?

Reply:
I was simply copying the code from Operate importer, but I agree, this looked wrong! I added update of the field.